saleor/plugins/vatlayer/plugin.py

```diff
@@ -206,6 +206,65 @@ def calculate_order_line_unit(
             variant.product, order_line.unit_price, country
         )
 
+    def get_checkout_line_tax_rate(
+        self,
+        checkout: "Checkout",
+        checkout_line_info: "CheckoutLineInfo",
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(checkout_line_info.product, address, previous_value)
+
+    def get_order_line_tax_rate(
+        self,
+        order: "Order",
+        product: "Product",
+        address: Optional["Address"],
+        previous_value: Decimal,
+    ) -> Decimal:
+        return self._get_tax_rate(product, address, previous_value)
+
+    def _get_tax_rate(
+        self, product: "Product", address: Optional["Address"], previous_value: Decimal
+    ):
+        if self._skip_plugin(previous_value):
+            return previous_value
+        country = address.country if address else None
+        taxes, tax_rate = self.__get_tax_data_for_product(product, country)
+        if not taxes or not tax_rate:
+            return previous_value
+        tax = taxes.get(tax_rate) or taxes.get(DEFAULT_TAX_RATE_NAME)
+        # tax value is given in precentage so it need be be converted into decimal value
+        return Decimal(tax["value"] / 100)
+
+    def get_checkout_shipping_tax_rate(
+        self,
+        checkout: "Checkout",
+        lines: Iterable["CheckoutLineInfo"],
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        previous_value: Decimal,
+    ):
+        return self._get_shipping_tax_rate(address, previous_value)
+
+    def get_order_shipping_tax_rate(self, order: "Order", previous_value: Decimal):
+        address = order.shipping_address or order.billing_address
```

Comment:
I think only `order.shipping_address` should be involved in this calculation. The billing address might be something totally different, wouldn't that result in a wrong calculation? E.g. for orders without shipping that don't have a shipping address?

Reply:
I agree with you, that it seems like only `order.shipping_address` should be taken into consideration. I did it in this way because the `calculate_order_shipping` is doing like that and both must work the same. We should ask @korycins why this is done like this.

Reply:
OK so if it's consistent with the previous implementation let's leave it for now and ping Maciek when he's back from the holidays.